engine/docker/docker.go

```diff
@@ -157,10 +157,31 @@ func (e *Docker) Destroy(ctx context.Context, pipelineConfig *spec.PipelineConfi
 	containers := e.containers
 	e.mu.Unlock()
 
+	timeout := time.Second * 30 // similar to k8s default timeout in pods
 	// stop all containers
 	for _, ctrName := range containers {
-		if err := e.client.ContainerKill(ctx, ctrName, "9"); err != nil {
-			logrus.WithField("container", ctrName).WithField("error", err).Warnln("failed to kill container")
+		if err := e.client.ContainerStop(ctx, ctrName, &timeout); err != nil {
+			logrus.WithField("container", ctrName).WithField("error", err).Warnln("failed to stop the container")
+		}
+
+		// Before removing the container we want to be sure that it's in a healthy state to be removed.
+
+		containerStatus, err := e.client.ContainerInspect(ctx, ctrName)
```

Comment:
Can maybe move these inside the for loop?

i.e. get status, if it is removing or running then sleep 1 and repeat, otherwise we're done.

Reply:
I was aiming for a do while. For a lot of containers, the state will be existed so we jump from 169 to 183 directly. For example, imagine a IACM pipeline, by the time the apply is running, the init and plan container already stopped (and exited) so when those containers are been inspected here, the state will be already exited